### ISSUE TITLE
Jib build is complicated

### ISSUE THREAD

**Issue by larsgrefer** [GoogleContainerTools/jib#1790]
While working on some other issues, I noticed that the build of jib itself is pretty complicated.

There seem to be 4 separate builds (3 gradle builds and one maven build) which are then glued together by the `build.sh` script.

Have you considered to just use one multi-module build?

**Comment by loosebazooka** [comment]
Yeah we've actually done some experiments to trigger gradle builds from maven, but it's not great. The thing is that you can't easily force maven to trigger external builds of dependencies AND also have them work well with the IDE. So what you end up with is worst case "no dependency found" or best case "out of date dependency". My current best solution is some sort of weird multi-module project where one module is responsible for triggering gradle builds and must be run before the actual jib-maven-plugin module and I'm not super fond of it.

Another option is to move the submodule builds to maven, and have gradle (which is more flexible) depend on maven projects. But maven builds are much slower and this is a much less desirable option.

On top of that, `build.sh` doesn't actually glue the projects together, it's just a convenience script. If you look through the build files, we're pulling source in directly from the subprojects, which is not great, but keeps us from having to make releases of the common libraries when we make changes to the plugins. Some of these problems are solvable with multimodule builds in combination with shade plugin.

How we do this is a little up in the air right now. It's something we've been meaning to figure out.

**Comment by larsgrefer** [comment]
I'd propose to just use one multi-module gradle build.

**Comment by loosebazooka** [comment]
Ah but `jib-maven-plugin` can't correctly order the building of its dependencies? Unless you're suggesting gradle trigger the maven build?

**Comment by larsgrefer** [comment]
I suggest to not use maven at all.

**Comment by loosebazooka** [comment]
I would be interested in a solution that continues to allow us to build `jib-maven-plugin`.

**Comment by larsgrefer** [comment]
There's a gradle plugin for building maven plugins with gradle: https://plugins.gradle.org/plugin/io.freefair.maven-plugin/2.9.5

See here for a maven-plugin which is built with it: https://github.com/joinfaces/joinfaces/tree/master/joinfaces-tools/joinfaces-maven-plugin

**Comment by loosebazooka** [comment]
Interesting. That would be nice. We'll take a look.

**Comment by larsgrefer** [comment]
I've opened #1791 so you can see how this change might look.

All members of this repository should have write access to the branch of my fork, so you can add changes to #1791 if you want.
